lib/ask/time_util.ex

```diff
@@ -1,16 +1,41 @@
 defmodule Ask.TimeUtil do
+  @months %{
+    1 => "Jan",
+    2 => "Feb",
+    3 => "Mar",
+    4 => "Apr",
+    5 => "May",
+    6 => "Jun",
+    7 => "Jul",
+    8 => "Aug",
+    9 => "Sep",
+    10 => "Oct",
+    11 => "Nov",
+    12 => "Dec",
+  }
+
   def format(ecto_timestamp, offset_seconds, offset_label) do
+    {{year, month, day}, {hour, min, sec}} = to_ymdhms(ecto_timestamp, offset_seconds)
+    "#{year}-#{pad2(month)}-#{pad2(day)} #{pad2(hour)}:#{pad2(min)}:#{pad2(sec)} #{offset_label}"
+  end
+
+  # Applies offset and formats datetime as `Timex.format!("%b %e, %Y %H:%M #{tz_offset}", :strftime)`.
+  # There are no reasons to not use Timex.format unless you need raw performance and make lots of
+  # repeating calls in a tight loop (eg. exporting a CSV) which is sole reason that method exists.
+  def format2(ecto_timestamp, offset_seconds, offset_label) do
+    {{year, month, day}, {hour, min, sec}} = to_ymdhms(ecto_timestamp, offset_seconds)
+    "#{Map.get(@months, month)} #{pad2(day, " ")}, #{year} #{pad2(hour)}:#{pad2(min)} #{offset_label}"
+  end
```

Comment:
It's awful to replace a couple readable calls into this, but the performance improvement is real when generating results: from over 5s to 2.1s when downloading a file on my local.